substrate/frame/revive/src/wasm/runtime.rs

```diff
@@ -1477,17 +1472,11 @@ pub mod env {
 	/// Stores the *free* balance of the current account into the supplied buffer.
 	/// See [`pallet_revive_uapi::HostFn::balance`].
 	#[api_version(0)]
-	fn balance(
-		&mut self,
-		memory: &mut M,
-		out_ptr: u32,
-		out_len_ptr: u32,
-	) -> Result<(), TrapReason> {
+	fn balance(&mut self, memory: &mut M, out_ptr: u32) -> Result<(), TrapReason> {
 		self.charge_gas(RuntimeCosts::Balance)?;
-		Ok(self.write_sandbox_output(
+		Ok(self.write_fixed_sandbox_output(
 			memory,
 			out_ptr,
-			out_len_ptr,
 			&self.ext.balance().encode(),
```

Comment:
In those cases we can also get rid of SCALE. It only was used because the type was generic and we didn't know the encoding. There are more occurrence.

```suggestion
			self.ext.balance().as_bytes(),
```

Reply:
only H256 has a as_bytes function, so made a custom free function `fn as_bytes(u: U256) -> [u8; 32]`